pkg/controller/external.go

```diff
@@ -166,21 +166,24 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 	if err != nil {
 		return managed.ExternalObservation{}, errors.Wrap(err, "cannot late initialize parameters")
 	}
-	lateInitialized := annotationsUpdated || lateInitedParams
-	resourceReady := tr.GetCondition(xpv1.TypeReady).Equal(xpv1.Available())
+	markedAvailable := tr.GetCondition(xpv1.TypeReady).Equal(xpv1.Available())
 	// We try to mark the resource ready before the spec late initialization
 	// and a call for "Plan" because those operations are costly, i.e. late-init
 	// causes a spec update which defers status update to the next reconcile and
 	// "Plan" takes a few seconds. We assume that a successful state refresh
 	// with state info available for the remote infrastructure in the output
 	// terraform.tfstate indicates readiness.
-	if !resourceReady || lateInitialized {
+	if !markedAvailable || annotationsUpdated || lateInitedParams {
```

Comment:
It took me quite a while to understand the reasoning of all the conditions in our offline chat and my old comment doesn't seem to be doing a good job there anymore. What do you think about a `switch` statement with cases for each of the conditions and whose `default:` is calling the plan & return? I think it'd end up in the same flow but just much easier to follow what's going on and why.